### PR TITLE
Pre-sealing holes

### PR DESCRIPTION
**Pull request by Kubuxu**
License: MIT
Signed-off-by: Jakub Sztandera <kubuxu@protocol.ai>

### CODE REVIEW

File: lib/sectorbuilder/sectorbuilder.go [hunk 1]
```diff
@@ -142,17 +144,21 @@ func New(cfg *Config, ds dtypes.MetadataDS) (*SectorBuilder, error) {
 	}
 
 	var lastUsedID uint64
-	b, err := ds.Get(lastSectorIdKey)
-	switch err {
-	case nil:
-		i, err := strconv.ParseInt(string(b), 10, 64)
-		if err != nil {
+	if cfg.OverrideLastID != 0 {
```
Comment (magik6k) [inline lib/sectorbuilder/sectorbuilder.go]: This feels hacky, I'd prefer writing that to the datastore when constructing sectorbuilder

File: lib/sectorbuilder/sectorbuilder.go [hunk 2]
```diff
@@ -119,12 +119,14 @@ type Config struct {
...
 
-	WorkerThreads uint8
+	WorkerThreads  uint8
+	OverrideLastID uint64
 
 	CacheDir    string
 	SealedDir   string
 	StagedDir   string
 	UnsealedDir string
+	_           struct{} // guard against nameless init
```
Comment (magik6k) [inline lib/sectorbuilder/sectorbuilder.go]: Meh, we aren't doing that anywhere

File: cmd/lotus-seed/main.go [hunk 3]
```diff
@@ -66,6 +66,11 @@ var preSealCmd = &cli.Command{
 			Value: 1,
 			Usage: "select number of sectors to pre-seal",
 		},
+		&cli.Uint64Flag{
+			Name:  "sector-offset",
+			Value: 0,
+			Usage: "select number of sectors to pre-seal",
```
Comment (magik6k) [inline cmd/lotus-seed/main.go]: Needs updating

**Comment by magik6k** [comment]
(tests need fixing too)

File: lib/sectorbuilder/sectorbuilder.go (hunk 2 above)
Comment (Kubuxu) [reply lib/sectorbuilder/sectorbuilder.go]: It was my way of verifying it.

File: lib/sectorbuilder/sectorbuilder.go (hunk 1 above)
Comment (Kubuxu) [reply lib/sectorbuilder/sectorbuilder.go]: I've changed it apply when nothing in the ds is found (and changed the name).

Comment (magik6k) [reply lib/sectorbuilder/sectorbuilder.go]: I'd still prefer writing to the datastore in preseal, but fine..

**Pull request merged**